### PR TITLE
Bean that has a bound interceptor must not declare final methods

### PR DESCRIPTION
**Pull request by mkouba**
- otherwise a deployment exception is thrown
- final methods declared on bean classes that require client proxies are
ignored though

### CODE REVIEW

**Review by geoand — APPROVED** [review]
LGTM

**Comment by mkouba** [comment]
Ach, "Imports are not sorted..." again...

**Review by manovotn — APPROVED** [review]

**Comment by gsmet** [comment]
Is it supposed to replace https://github.com/quarkusio/quarkus/pull/5104 ?

**Comment by geoand** [comment]
No, it's a prerequisite. I have rebased that PR onto this one locally so once this one is ready, I'll force push the to #5104

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java [hunk 1]
```diff
@@ -434,7 +438,14 @@ protected String getType() {
                 }
             }
 
-            Methods.addInterceptedMethodCandidates(beanDeployment, target.get().asClass(), candidates, classLevelBindings);
+            Set<MethodInfo> finalMethods = Methods.addInterceptedMethodCandidates(beanDeployment, target.get().asClass(),
+                    candidates, classLevelBindings);
+            if (!finalMethods.isEmpty()) {
+                errors.add(new DeploymentException(String.format(
+                        "Bean %s has a bound interceptor and must not declare final methods:\n\t- %s", getBeanClass(),
+                        finalMethods.stream().map(Object::toString).collect(Collectors.joining("\n\t- ")))));
```
Comment (gsmet) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java]: Minor nitpick but could you add a `sorted()` after the `map()` call? Having a deterministic order helps when you go through the error message to fix the issue.

**Comment by mkouba** [comment]
@gsmet added `FinalNonInterceptedMethodTest` and changed the wording a little...

**Review by gsmet — APPROVED** [review]

**Comment by manovotn** [comment]
`Imports are not sorted `
@mkouba looks like you've commited another sorting sin ;)

**Comment by geoand** [comment]
Imports again :P

**Comment by mkouba** [comment]
Eh, I need to import more coffee. Or maybe start doing something meaningful ;-).

**Comment by gsmet** [comment]
And merged, thanks @mkouba and @geoand !

**Pull request merged**